### PR TITLE
Add Balance Component

### PR DESCRIPTION
**Pull request by sulfurious**
- Refactor modules to use selectors
- Add Balance component underneath grid

### CODE REVIEW

**Comment by jmalfatto** [comment]
@sulfurious I know you were just following the wireframe design, but I pointed out in our first planning meeting that the design of the boxes containing the balance numbers screams out to me "editable", when in fact they are read-only. Not saying this version looks great (see screenshot below), but simply turning off the `border` and `background-color` properties on the `.balance-amount` class appears to solve that problem.

screen shot 2017-03-14 at 10 26 28 am

File: client/components/Balance/BalanceRow.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React, { PropTypes } from 'react';
+
+import './style.scss';
+
+const Balance = ({ children }) => {
+  return (
+    <div className="balance-row-container">
```
Comment (grgur) [inline client/components/Balance/BalanceRow.js]: soooo why are we nesting here?

File: client/components/Balance/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { PropTypes } from 'react';
+
+import formatAmount from 'utils/formatAmount';
+import './style.scss';
+
+const Balance = ({ title, symbol, amount, colorize }) => {
+  const formattedAmount = formatAmount(amount, false);
```
Comment (grgur) [inline client/components/Balance/index.js]: ahhh I've been waiting for this 😃 . 
this is kinda business logic, isn't it. If you pre-format the amount in redux OR if you use reselect for obtaining amount, you'd avoid recalculating formatting on every render 😉

File: client/containers/Balance/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import {
+  getBalance,
+  getInflowBalance,
+  getOutflowBalance
+} from 'modules/transactions';
+
+import BalanceItem from 'components/Balance';
+import BalanceRow from 'components/Balance/BalanceRow';
+
+@connect(
```
Comment (grgur) [inline client/containers/Balance/index.js]: spot on!

File: client/containers/Balance/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    outflow: PropTypes.string.isRequired
+  }
+
+  render() {
+    const { inflow, outflow, balance } = this.props;
+
+    return (
+      <BalanceRow>
+        <BalanceItem amount={inflow} title="Total Inflow" />
+        <BalanceItem amount={outflow} title="Total Outflow" symbol="-" />
```
Comment (grgur) [inline client/containers/Balance/index.js]: I'd use a component for showing prefix (symbol). It feels more intuitive, but maybe that's just me.

File: client/modules/transactions.js [hunk 5]
```diff
@@ -55,18 +46,55 @@ function addTransactionToState(state, action) {
   return newState;
 }
 
+function totalTransactions(transactions) {
+  return transactions.reduce((total, item) => total + parseFloat(item.value), 0).toFixed(2);
+}
+
+/**
+ * Selectors
```
Comment (grgur) [inline client/modules/transactions.js]: I ❤️ this!

File: client/components/Balance/BalanceRow.js (hunk 1 above)
Comment (sulfurious) [inline client/components/Balance/BalanceRow.js]: Need to for styling. balance-row sizes itself to its content width. balance-row-container centers it. If we could get away with a gray box that's the full width, we could get rid of a div but tat's what the wireframe shows.

File: client/components/Balance/index.js (hunk 2 above)
Comment (sulfurious) [inline client/components/Balance/index.js]: I don't see any business logic here. It's just formatting but I agree that if we can save cycles on rendering then we should. I'll whip up a selector for this.

File: client/containers/Balance/index.js (hunk 4 above)
Comment (sulfurious) [inline client/containers/Balance/index.js]: Done

**Pull request merged**